src/controllers/vdeployment_controller/trusted/util.rs

```diff
@@ -39,10 +39,16 @@ pub open spec fn filter_old_and_new_vrs(vd: VDeploymentView, vrs_list: Seq<VRepl
         &&& match_template_without_hash(vd.spec.template, vrs)
         &&& vrs.spec.replicas is None || vrs.spec.replicas.unwrap() > 0
```

Comment:
I suggest adding a comment to explain that line 40 says that the `vrs.spec.replicas` is effectively not 0, and mention the default behavior of ReplicaSet that interprets empty replicas field as 1. Also, don't forget that we need to fix it in VRS controller.